futures/src/lib.rs

```diff
@@ -140,21 +140,34 @@ pub mod future {
     //!   immediate defined value.
 
     pub use futures_core::future::{
-        FutureOption, Future, TryFuture, ReadyFuture, ready,
+        FutureOption, Future, TryFuture,
         FutureObj, LocalFutureObj, UnsafeFutureObj,
+        ready, ReadyFuture,
     };
+
     pub use futures_util::future::{
-        Empty, Flatten, FlattenStream, Fuse, Inspect, IntoStream, Lazy,
-        Then, PollFn, Map, FutureExt, empty, lazy, poll_fn,
-        // AndThen, ErrInto, Join, Join3, Join4, Join5, LoopFn,
-        // MapErr, OrElse, Select, Loop, loop_fn, Either
+        empty, Empty,
+        lazy, Lazy,
+        maybe_done, MaybeDone,
+        poll_fn, PollFn,
+
+        FutureExt,
+        FlattenStream, Flatten, Fuse, Inspect, IntoStream, Join, Join3, Join4,
+        Join5, Map, Then, WithExecutor,
     };
 
     #[cfg(feature = "std")]
     pub use futures_util::future::{
-        CatchUnwind
-        // JoinAll, SelectAll, SelectOk, Shared, SharedError, SharedItem,
-        // join_all, select_all, select_ok
+        // For FutureExt:
+        CatchUnwind, Shared
+
+        // ToDo: JoinAll, SelectAll, SelectOk, join_all, select_all, select_ok
+    };
+
+    pub use futures_util::try_future::{
+        TryFutureExt,
+        AndThen, ErrInto, FlattenSink, IntoFuture, MapErr, MapOk, OrElse,
+        UnwrapOrElse,
```

Comment:
I've put the stuff for `TryFuture` into the `future` module

Reply:
Currently the `TryFuture` trait does not have `Future` as super trait. But, this is planned to change  as soon as it becomes technically possible. Therefore, merging the two makes sense to me. What do you think?